src/ucp/wireup/select.c

```diff
@@ -636,7 +648,10 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_add_lane_desc(
         }
     }
 
-    if (select_ctx->num_lanes >= UCP_MAX_LANES) {
+    max_lanes = ucp_wireup_is_fast_path_lane(lane_type) ?
+                        UCP_MAX_FAST_PATH_LANES :
+                        UCP_MAX_LANES;
+    if (select_ctx->num_lanes >= max_lanes) {
```

Comment:
if we don't rely on adding fast path lanes first,
we need to count fast_path and slow_path lanes separately and then "sort" them so that fast will be first.
if we rely on it - need to document (here) and maybe add some asserts (that we don't add fast-path lane after already having slow-path lanes)